numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1653,23 +1653,15 @@ array_new(PyTypeObject *subtype, PyObject *args, PyObject *kwds)
         if (is_f_order) {
             buffer.flags |= NPY_ARRAY_F_CONTIGUOUS;
         }
-        ret = (PyArrayObject *)\
-            PyArray_NewFromDescr_int(subtype, descr,
-                                     dims.len, dims.ptr,
-                                     strides.ptr,
-                                     offset + (char *)buffer.ptr,
-                                     buffer.flags, NULL, 0, 1);
+        ret = (PyArrayObject *)PyArray_NewFromDescr_int(
+                subtype, descr,
+                dims.len, dims.ptr, strides.ptr, offset + (char *)buffer.ptr,
+                buffer.flags, NULL, buffer.base,
```

Comment:
This is the only call where `obj` is not the same as `base`, and thus the reason for the change in signature of `PyArray_NewFromDescr_int`. I wonder if this particular change is necessary/helpful, since in the other cases you can now guarantee that, inside `__array_finalize__`, `obj is self.base`. If you think it is helpful, might it make sense to just add another flag to `PyArray_NewFromDescr`, say, `use_base_as_obj`.

Reply:
Is `__array_finalize__` always supposed to be called with an `ndarray`? Maybe we should just pass this through too

Reply:
The [documentation](http://www.numpy.org/devdocs/user/c-info.beyond-basics.html#ndarray.__array_finalize__) specifically states `sub-type`

Reply:
> This is the only call where obj is not the same as base

I was about to thank you for that observation, but unfortunately there are a few more cases where that's not true.